python-frontend/src/main/java/org/sonar/python/semantic/SymbolImpl.java

```diff
@@ -132,4 +153,16 @@ public void removeUsages() {
   Map<String, Symbol> getChildrenSymbolByName() {
     return Collections.unmodifiableMap(childrenSymbolByName);
   }
+
+  @Nullable
+  public static Symbol getTypeSymbolFromExpression(Expression expression) {
```

Comment:
do we need this to be public? What about having `setAnnotatedTypeName` taking a TypeAnnotation tree instead as parameter and calling this function?